src/server/saved_objects/client/lib/create_find_query.js

```diff
@@ -35,5 +40,20 @@ export function createFindQuery(options = {}) {
     });
   }
 
-  return { version: true, query: { bool } };
+  const query = { version: true, query: { bool } };
+
+  if (sortField) {
+    const value = {
+      order: sortOrder,
+      unmapped_type: get(mappings, `${type}.properties.${sortField}.type`)
```

Comment:
You're right - found this when working on getting Kibana to work against ES master. Updating the mapping definition is definitely what caused it.